blockmanager.go

```diff
@@ -1048,6 +1054,10 @@ func (b *blockManager) handleInvMsg(imsg *invMsg) {
 				numRequested++
 			}
 
+			if !blocksRequested {
```

Comment:
I'd just do `blocksRequested = true` here.  There is unlikely to be any efficiency difference, but it would be a little cleaner I think.